src/architecture/unixfs.md

```diff
@@ -0,0 +1,618 @@
+`/` (`0x2f`) unicode codepoints however CIDs may use a multibase encoding with
+a `/` in the alphabet if the encoded CID does not contain `/` once encoded.
+
+Everything following the CID is a collection of path component (some bytes)
+separated by `/` (`0x2F`). UnixFS paths read from left to right, and are
+inspired by POSIX paths.
+
+- Components MUST NOT contain `/` unicode codepoints because else it would break
+  the path into two components.
+- Components SHOULD be UTF8 unicode.
+- Components are case sensitive.
+
+### Escaping
+
+The `\` may be supposed to trigger an escape sequence. However, it is currently
+broken and inconsistent across implementations. Until we agree on a specification
+for this, you SHOULD NOT use any escape sequences and non-ASCII characters.
+
+### Relative Path Components
+
+Relative path components MUST be resolved before trying to work on the path:
+
+- `.` points to the current node, those path components MUST be removed.
+- `..` points to the parent node, they MUST be removed left to right. When removing
+  a `..`, the path component on the left MUST also be removed. If there is no path
+  component on the left, you MUST error since it is an attempt of out-of-bounds
+  path resolution.
+
+### Restricted Names
+
+The following names SHOULD NOT be used:
+
+- The `.` string: represents the self node in POSIX pathing.
+- The `..` string: represents the parent node in POSIX pathing.
+- The empty string. <!--TODO: check that this is true-->
+- Any string containing a `NUL` (`0x00`) byte: this is often used to signify string
+  terminations in some systems (such as most C compatible systems), and many unix
+  file systems do not accept this character in path components.
+
+## Design Decision Rationale
+
+### `mtime` and `mode` Metadata Support in UnixFSv1.5
+
+Metadata support in UnixFSv1.5 has been expanded to increase the number of possible
+use cases. These include rsync and filesystem based package managers.
+
+Several metadata systems were evaluated, as discussed in the following sections.
+
+#### Separate Metadata Node
+
+In this scheme, the existing `Metadata` message is expanded to include additional
+metadata types (`mtime`, `mode`, etc). It contains links to the actual file data
+but never the file data itself.
+
+This was ultimately rejected for a number of reasons:
+
+1. You would always need to retrieve an additional node to access file data which
+  limits the kind of optimizations that are possible. For example many files are
+  under the 256 KiB block size limit, so we tend to inline them into the describing
+  UnixFS `File` node. This would not be possible with an intermediate `Metadata` node.
+2. The `File` node already contains some metadata (e.g. the file size) so metadata
+  would be stored in multiple places which complicates forwards compatibility with
+  UnixFSv2 as to map between metadata formats potentially requires multiple fetch
+  operations.
```

Comment:
```suggestion
In this scheme, the existing `Metadata` message is expanded to include additional
metadata types (`mtime`, `mode`, etc). It contains links to the actual file data,
but never the file data itself.

This was ultimately rejected for a number of reasons:

1. You would always need to retrieve an additional node to access file data, which
  limits the kind of optimizations that are possible. For example, many files are
  under the 256 KiB block size limit, so we tend to inline them into the describing
  UnixFS `File` node. This would not be possible with an intermediate `Metadata` node.
2. The `File` node already contains some metadata (e.g. the file size), so metadata
  would be stored in multiple places. This complicates forwards compatibility with
  UnixFSv2, as mapping between metadata formats potentially requires multiple fetch
  operations.
```